src/software/ai/navigator/obstacle/obstacle_test.cpp

```diff
@@ -3,231 +3,120 @@
-        Point pt5 = Point(4.034, 4.273);
-        EXPECT_DOUBLE_EQ(pg.getPoints().size(), 6);
-        EXPECT_NEAR(pg.getPoints()[0].x(), pt0.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[0].y(), pt0.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].x(), pt1.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].y(), pt1.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].x(), pt2.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].y(), pt2.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
-}
+    Polygon expected = Polygon({
+        {-1, -3},
+        {-1, 1},
+        {2, 1},
+        {2, -3},
+    });
 
-// obstacle with robot radius factor = 1.2, velocity projection factor = 1.4
-// centred at (-1,-2) and with (-3,2) velocity
-TEST(NavigatorObstacleTest, shifted_scaled_up_velocity_obstacle_polygon)
-{
-    double TEST_EPSILON    = 1e-3;
-    Timestamp current_time = Timestamp::fromSeconds(123);
-    Robot robot    = Robot(3, Point(-1, -2), Vector(-3, 2), Angle::fromRadians(2.2),
-                        AngularVelocity::fromRadians(-0.6), current_time);
-    auto p_pointer = Obstacle::createRobotObstacleWithScalingParams(robot, 1.2, 1.4)
-                         .getBoundaryPolygon();
-
-    EXPECT_TRUE(p_pointer);
-
-    if (p_pointer)
-    {
-        Polygon pg = *p_pointer;
-        // visually verified
-        Point pt0 = Point(-1.138, -2.208);
-        Point pt1 = Point(-.889, -2.224);
-        Point pt2 = Point(-0.751, -2.016);
-        Point pt3 = Point(-0.862, -1.792);
-        Point pt4 = Point(-5.241, 1.127);
-        Point pt5 = Point(-5.518, 0.712);
-        EXPECT_DOUBLE_EQ(pg.getPoints().size(), 6);
-        EXPECT_NEAR(pg.getPoints()[0].x(), pt0.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[0].y(), pt0.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].x(), pt1.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].y(), pt1.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].x(), pt2.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].y(), pt2.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    EXPECT_EQ(expected.getPoints(), polygon_obstacle.getPolygon().getPoints());
```

Comment:
Maybe worth adding an equality operator for Polygons? (perhaps a separate issue)

Reply:
I just went ahead and did it since it's not a lot more code

Reply:
not using the new polygon equality here

Reply:
fixed